src/language-html/print/block.js

```diff
@@ -0,0 +1,123 @@
+import { group, hardline, indent } from "../../document/builders.js";
+import { printChildren } from "./children.js";
+
+let uid = 0;
+
+const settings = {
```

Comment:
Please use `const settings = Object.assign(Object.create(null), { /* ... */ })`. For appropriate error messages for `@constructor` and `@toString`.

**Prettier pr-15606**
[Playground link](https://deploy-preview-15606--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEABGEDKMBOBLKAcwAIAKASmOAF8AdKEAGhAgAcY9oBnZUAQxw4IAdwAKAhDxR8ANsL4BPHswBGOPmADWcGJj4BbOABkCcZADNZXOKvVadmVhoKFkuAK42Q1-Xjc5PZi4XGTgARXcIeAsrLwArLgAPTBDwyOikSxlrZgBHdLhRIVYpED4uAFooODgAEzqmEFw+PBkXAGEIfX0+ZDKZGUbgolCAQRhcPBV3eFE4HBNqmOyvAAsYfRkAdVW8eC4nMDhMST28ADc9hT6wLmUQc88ASSh62EwwfHZR1+wFUOWORArCE1i26lYfRBcGsOHOZmYBFhMCKfEIPUBXicOFhfT4RHcMgEjRBBBgWzwtRgq2QAA4AAzMHBwfJ4Zmo9G9TKxZgwPgqClUmlIABMzHc1gAKvypFkgXB9Co6vVakZ8YR3Gi4AAxCA4HoTFx4mYQEDUahAA)
<!-- prettier-ignore -->
```sh
--parser angular
```

**Input:**
<!-- prettier-ignore -->
```html
@toString () {}

```

**Output:**
<!-- prettier-ignore -->
```html
TypeError: Cannot read properties of undefined (reading 'length')
    at Ea (https://deploy-preview-15606--prettier.netlify.app/lib/plugins/html.js:14:11463)
    at is (https://deploy-preview-15606--prettier.netlify.app/lib/plugins/html.js:14:11350)
    at Object.ba [as print] (https://deploy-preview-15606--prettier.netlify.app/lib/plugins/html.js:14:12544)
    at Yn (https://deploy-preview-15606--prettier.netlify.app/lib/standalone.js:30:7727)
    at D (https://deploy-preview-15606--prettier.netlify.app/lib/standalone.js:30:7547)
    at s (https://deploy-preview-15606--prettier.netlify.app/lib/standalone.js:30:7355)
    at rt (https://deploy-preview-15606--prettier.netlify.app/lib/plugins/html.js:14:7858)
    at https://deploy-preview-15606--prettier.netlify.app/lib/plugins/html.js:14:8860
    at https://deploy-preview-15606--prettier.netlify.app/lib/standalone.js:26:1596
    at Vt.each (https://deploy-preview-15606--prettier.netlify.app/lib/standalone.js:26:1499)
```

Reply:
Why don't we use a `Map` instead.

Reply:
Either is fine for me!

Reply:
Has been replaced by `Map`